paimon-core/src/main/java/org/apache/paimon/mergetree/compact/FirstRowMergeFunctionWrapper.java

```diff
@@ -66,11 +66,7 @@ public ChangelogResult getResult() {
             return reusedResult;
         }
 
-        reusedResult.setResult(result);
-        if (result.level() == 0) {
```

Comment:
Why removing this `if`? What about merging level 4 and level 5?

Reply:
if merging with high level, it should return at line 61.